### PR TITLE
Package updates for slate, lapackpp, and blaspp

### PR DESCRIPTION
**Pull request by G-Ragghianti**
These package updates correspond to SLATE release 2022-07-00.  The major change is the inclusion of HIP and CUDA support in lapackpp.

### CODE REVIEW

File: var/spack/repos/builtin/packages/lapackpp/package.py [hunk 1]
```diff
@@ -52,16 +56,34 @@ class Lapackpp(CMakePackage):
     for (lpp_ver, bpp_ver) in _versions:
         depends_on("blaspp@" + bpp_ver, when="@" + lpp_ver)
 
+    depends_on("blaspp ~cuda", when="~cuda")
+    depends_on("blaspp +cuda", when="+cuda")
+    depends_on("blaspp ~rocm", when="~rocm")
```
Comment (mgates3) [inline var/spack/repos/builtin/packages/lapackpp/package.py:61]: Does `lapackpp~cuda` actually require `blaspp~cuda`,
and `lapackpp~rocm` require `blaspp~rocm`?
Normally that's what I would expect, that both are `+cuda`, or both `+rocm`, or both `~rocm ~cuda`, but I don't think it's required. It could be `lapackpp~rocm~cuda` and (`blaspp+rocm` or `blaspp+cuda`). But those are edge cases I wouldn't expect and wouldn't test, so may be simpler to disallow.

File: var/spack/repos/builtin/packages/slate/package.py [hunk 2]
```diff
@@ -54,9 +57,14 @@ class Slate(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("blaspp ~cuda", when="~cuda")
     depends_on("blaspp +cuda", when="+cuda")
     depends_on("blaspp ~rocm", when="~rocm")
+    depends_on("lapackpp ~cuda", when="~cuda")
+    depends_on("lapackpp +cuda", when="+cuda")
+    depends_on("lapackpp ~rocm", when="~rocm")
```
Comment (mgates3) [inline var/spack/repos/builtin/packages/slate/package.py:62]: Ditto: `slate~cuda` doesn't necessarily imply `lapackpp~rocm` and `blaspp~rocm`. Again, an edge case that maybe isn't important.

**Review by mgates3 — APPROVED** [review]
Looks good. Just the one question about negative dependencies (`~cuda` requiring `~cuda`, `~rocm` requiring `~rocm`), which isn't a big deal.

**Review by tldahlgren — APPROVED** [review]
Confirmed all three new sha256.

File: var/spack/repos/builtin/packages/lapackpp/package.py (hunk 1 above)
Comment (G-Ragghianti) [reply var/spack/repos/builtin/packages/lapackpp/package.py:61]: I'd say that since we don't test those edge cases and they are unusual, that those cases should be considered unsupported.  It is more likely that someone would want cuda or rocm support turned on for all major components, so this makes it much easier for the user by chaining the gpu config down to the dependencies.

**Comment by tldahlgren** [comment]
@G-Ragghianti @mgates3 Let me know when you're ready for this to be merged.

File: var/spack/repos/builtin/packages/slate/package.py (hunk 2 above)
Comment (G-Ragghianti) [reply var/spack/repos/builtin/packages/slate/package.py:62]: If someone really needs to do that and we want to support it, it could be done, but it would complicate the spack package and would make it more difficult for the users who just want to turn on or off cuda nad rocm support for all components (slate, lapackpp, and blaspp).

Comment (G-Ragghianti) [reply var/spack/repos/builtin/packages/slate/package.py:62]: One addition: this just replicates the previous behavior such that slate depends on lapackpp in the same way that slate has previously depended on blaspp.

**Comment by mgates3** [comment]
@tldahlgren Everything looks good. Thanks.
